_posts/2018-11-06-simple-slackbot.md

```diff
@@ -0,0 +1,55 @@
+---
+layout: post
+title:  "Simple Slackbot"
+date: 2018-11-06 00:03:30 +0200
+categories: javascript
+author_name: Sergii Paryzhskyi
+author_url : /author/sergii_paryzhskyi
+author_avatar: sergii_paryzhskyi
+read_time : 7
+show_related_posts: false
+square_related: recommend-wolf
+feature_image: posts/2018-11-06-simple-slackbot/poster.jpg
+---
+
+Recently I've built a small slackbot that is posting with certain regularity in specified slack-channel. I will tell you about the idea behind it and also share some technical details.
+
+
+## The Idea
+
+The idea to build such a bot came from one colleague of mine - [Jacek Smolak][jacek]. At that point he was reading the book ["97 Things Every Programmer Should Know"][book-link]. In order to share those programmer tips with others and also remind them to himself, he came up with idea of posting each of them one by one on a daily basis in appropriate slack channel. For us this slack channel is #craft, you probably have something similar amoung your channels in slack.
+
+
+## Implementation
+
+The implementation is dead simple. As it was written with JavaScript, the easiest way was to use an official [slack npm package][slack-npm]. Posting something in specific channel becoming a one-liner that returns a Promise that we can handle depending if it is success or failure:
+
+```js
+slackClient.chat.postMessage({
+    channel: '#craft',
+    message: '...'
+});
+```
+
+The very same interface is used for sending direct messages, so without changing anything you can specify your own slack-name instead of channel. In this case you will be getting the messages from the bot directly. Its quite handy for testing purposes as well.
+Other thing that needed to be done is that we don't want to pick a random quote from json every time we post, but rather to relate those quotes to specific days. In other words, every day has its own quote and doesn't matter how often we run script during the day, this quote will be the only one for this day. That was easy to achieve by picking a quote based on day of the year modulo the amount of quotes in picked json file, so we want pick something that is outside of the scope.
```

Comment:
json is mentioned but not explained:
 - file / structure
 - where did that came from as well?

I would explain that for the purpose of posting the quotes we need to store them somewhere. Be it a backend service or a file. I would then mention the stuff you wrote.